pkg/cmd/run/view/view.go

```diff
@@ -322,9 +318,18 @@ func runView(opts *ViewOptions) error {
 		}
 		defer runLogZip.Close()
 
-		attachRunLog(&runLogZip.Reader, jobs)
+		zlm := getZipLogMap(&runLogZip.Reader, jobs)
+		segments := populateLogSegments(jobs, zlm, opts.LogFailed)
 
-		return displayRunLog(opts.IO.Out, jobs, opts.LogFailed)
+		err = assignMissingLogs(segments, httpClient, repo, run.ID)
+		if err != nil {
+			if errors.Is(err, errTooManyAPILogFetchers) {
+				return fmt.Errorf("too many API requests needed to fetch logs; try narrowing down to a specific job with the `--job` option")
+			}
+			return err
+		}
+
+		return displayLogSegments(opts.IO.Out, segments)
```

Comment:
@williammartin, this is actually the overall look of the new implementation:

1. Make a map of the log files available in the ZIP archive (the `logMap` we created earlier in this PR, is just now about the ZIP archive content).
2. A slice of `logSegment`s is put together based on the log files available in the ZIP archive. a `logSegment` can be an individual step log, or and entire job log.
3. Any `logSegment` with an unassigned `fetcher` field, will be assigned with an `apiLogFetcher`. Here we also apply the restriction on the count (which is 25) and error out if the number exceeds.
4. Display the `logSegments`.